processor/attributesprocessor/factory.go

```diff
@@ -131,9 +146,17 @@ func buildAttributesConfiguration(config Config) ([]attributeAction, error) {
 			} else {
 				action.FromAttribute = a.FromAttribute
 			}
+			if a.Regex != "" {
+				re, err := regexp.Compile(a.Regex)
+				if err != nil {
+					return nil, fmt.Errorf("error creating \"attributes\" processor. Field \"regex\" has invalid pattern: \"%s\" to be set at the %d-th actions of processor %q", a.Regex, i, config.Name())
+				}
+				action.Regex = re
+				action.AttrNames = re.SubexpNames()
```

Comment:
I'd consider raising:

- A warning if any `AttrNames` are an empty string (i.e. the regex included an un-named & un-ignored capture group)
- An error if no `AttrNames` are a non-empty string (i.e. there are no named capture groups in the regex)

Reply:
Added this extra validation along with unit tests for those conditions.